upgrade/upgrades: TestBackfillJobsInfoTable failed

upgrade/upgrades.TestBackfillJobsInfoTable [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9260292?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9260292?buildTab=artifacts#/) on master @ [d107217dac5d817cc115bc0e97b7e53c0f2878bf](https://github.com/cockroachdb/cockroach/commits/d107217dac5d817cc115bc0e97b7e53c0f2878bf):

Fatal error:

```
panic: unexpectedly no result routers in {0xc012dd6e00 [] [] {[]} 0 local} [recovered]
	panic: unexpectedly no result routers in {0xc012dd6e00 [] [] {[]} 0 local}

```
Stack: 

```
goroutine 8592023 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x1e1868a?, {0x5ce66e8, 0xc00472bef0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:229 +0x6a
panic({0x46156c0, 0xc0087ee520})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/sql/physicalplan.(*PhysicalPlan).GetResultTypes(0xc0017222a0?)
	github.com/cockroachdb/cockroach/pkg/sql/physicalplan/physical_plan.go:226 +0xba
github.com/cockroachdb/cockroach/pkg/sql/physicalplan.(*PhysicalPlan).EnsureSingleStreamOnGateway(0xc0133d0580, {0x5ce66e8, 0xc008f76420})
	github.com/cockroachdb/cockroach/pkg/sql/physicalplan/physical_plan.go:494 +0x7c
github.com/cockroachdb/cockroach/pkg/sql.finalizePlanWithRowCount({0x5ce66e8, 0xc008f76420}, 0x0?, 0xc0133d0580, 0x0?)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:4684 +0x6e
github.com/cockroachdb/cockroach/pkg/sql.FinalizePlan(...)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:4671
github.com/cockroachdb/cockroach/pkg/sql/importer.distImport.func1({0x5ce66e8, 0xc008f76420}, 0x4a67c0f?)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor_planning.go:115 +0x5cf
github.com/cockroachdb/cockroach/pkg/sql/importer.distImport({_, _}, {_, _}, _, _, {_, _, _}, {0x0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor_planning.go:119 +0x2f9
github.com/cockroachdb/cockroach/pkg/sql/importer.ingestWithRetry({_, _}, {_, _}, _, _, {_, _, _}, {0x0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_job.go:1258 +0x76d
github.com/cockroachdb/cockroach/pkg/sql/importer.(*importResumer).Resume(0xc00f221360, {0x5ce66e8, 0xc00ee836b0}, {0x4803040?, 0xc0014b1f38})
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_job.go:290 +0xb26
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0xc00e2d11e0?, 0x0?, {0x5cbfdd0?, 0xc00f221360?}, {0x5ce66e8?, 0xc00ee836b0?}, {0x4803040?, 0xc0014b1f38?}, 0xc001723860)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1534 +0xcc
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc00377cd80, {0x5ce66e8, 0xc012de5c50}, {0x4803040, 0xc0014b1f38}, {0x5cbfdd0, 0xc00f221360}, 0xc00e2d11e0, {0xc0092f7768, 0x7}, ...)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1535 +0x8ed
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc00377cd80, {0x5ce6640, 0xc00f211740}, {0x5cbfdd0, 0xc00f221360}, 0xc00e2d11e0, {0xc0092f7768, 0x7}, {0xc00754c670, 0x5})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:467 +0x5a6
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func2({0xc00f266800?, 0xc00885b980?})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:385 +0xfa
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470 +0x146
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:461 +0x43b
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestBackfillJobsInfoTable
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/194b1b11adb2bb86a3c51c4924595952/logTestBackfillJobsInfoTable299676074
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-schema
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillJobsInfoTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

upgrade/upgrades.TestBackfillJobsInfoTable [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9268756?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9268756?buildTab=artifacts#/) on master @ [2bd2c806ab3044569b09e0a205b5bc0452ad4e2b](https://github.com/cockroachdb/cockroach/commits/2bd2c806ab3044569b09e0a205b5bc0452ad4e2b):

Fatal error:

```
panic: unexpectedly no result routers in {0xc0075c8070 [] [] {[]} 0 local} [recovered]
	panic: unexpectedly no result routers in {0xc0075c8070 [] [] {[]} 0 local}

```
Stack: 

```
goroutine 8577964 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x1e1868a?, {0x5cec908, 0xc00e4fdad0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:229 +0x6a
panic({0x4619480, 0xc00f1ceb10})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/sql/physicalplan.(*PhysicalPlan).GetResultTypes(0xc0116122a0?)
	github.com/cockroachdb/cockroach/pkg/sql/physicalplan/physical_plan.go:226 +0xba
github.com/cockroachdb/cockroach/pkg/sql/physicalplan.(*PhysicalPlan).EnsureSingleStreamOnGateway(0xc000a5be80, {0x5cec908, 0xc0145c3a70})
	github.com/cockroachdb/cockroach/pkg/sql/physicalplan/physical_plan.go:494 +0x7c
github.com/cockroachdb/cockroach/pkg/sql.finalizePlanWithRowCount({0x5cec908, 0xc0145c3a70}, 0x0?, 0xc000a5be80, 0x0?)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:4684 +0x6e
github.com/cockroachdb/cockroach/pkg/sql.FinalizePlan(...)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:4671
github.com/cockroachdb/cockroach/pkg/sql/importer.distImport.func1({0x5cec908, 0xc0145c3a70}, 0x4a6bc6f?)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor_planning.go:115 +0x5cf
github.com/cockroachdb/cockroach/pkg/sql/importer.distImport({_, _}, {_, _}, _, _, {_, _, _}, {0x0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor_planning.go:119 +0x2f9
github.com/cockroachdb/cockroach/pkg/sql/importer.ingestWithRetry({_, _}, {_, _}, _, _, {_, _, _}, {0x0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_job.go:1258 +0x76d
github.com/cockroachdb/cockroach/pkg/sql/importer.(*importResumer).Resume(0xc006012690, {0x5cec908, 0xc00c74cff0}, {0x4806d40?, 0xc008ce6178})
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_job.go:290 +0xb26
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0xc00f13e1a0?, 0x0?, {0x5cc5ff0?, 0xc006012690?}, {0x5cec908?, 0xc00c74cff0?}, {0x4806d40?, 0xc008ce6178?}, 0xc011613860)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1534 +0xcc
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc01478b200, {0x5cec908, 0xc0104e0ab0}, {0x4806d40, 0xc008ce6178}, {0x5cc5ff0, 0xc006012690}, 0xc00f13e1a0, {0xc002c029b0, 0x7}, ...)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1535 +0x8ed
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc01478b200, {0x5cec860, 0xc0078b3d80}, {0x5cc5ff0, 0xc006012690}, 0xc00f13e1a0, {0xc002c029b0, 0x7}, {0xc002c02c70, 0x5})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:467 +0x5a6
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func2({0xc00c339c00?, 0xc0084c9a10?})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:385 +0xfa
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470 +0x146
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:461 +0x43b
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestBackfillJobsInfoTable
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/194b1b11adb2bb86a3c51c4924595952/logTestBackfillJobsInfoTable1301047440
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillJobsInfoTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

upgrade/upgrades.TestBackfillJobsInfoTable [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9268760?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9268760?buildTab=artifacts#/) on master @ [2bd2c806ab3044569b09e0a205b5bc0452ad4e2b](https://github.com/cockroachdb/cockroach/commits/2bd2c806ab3044569b09e0a205b5bc0452ad4e2b):

Fatal error:

```
panic: unexpectedly no result routers in {0xc004640070 [] [] {[]} 0 local} [recovered]
	panic: unexpectedly no result routers in {0xc004640070 [] [] {[]} 0 local}

```
Stack: 

```
goroutine 52341 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x4afa45?, {0xa322108, 0xc00041af60})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:229 +0x79
panic({0x6d0ec80, 0xc00243e410})
	GOROOT/src/runtime/panic.go:890 +0x262
github.com/cockroachdb/cockroach/pkg/sql/physicalplan.(*PhysicalPlan).GetResultTypes(0xc005456e00)
	github.com/cockroachdb/cockroach/pkg/sql/physicalplan/physical_plan.go:226 +0x1c5
github.com/cockroachdb/cockroach/pkg/sql/physicalplan.(*PhysicalPlan).EnsureSingleStreamOnGateway(0xc005456e00, {0xa322108, 0xc000d17680})
	github.com/cockroachdb/cockroach/pkg/sql/physicalplan/physical_plan.go:494 +0x1af
github.com/cockroachdb/cockroach/pkg/sql.finalizePlanWithRowCount({0xa322108, 0xc000d17680}, 0x0?, 0xc005456e00, 0x0?)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:4684 +0x94
github.com/cockroachdb/cockroach/pkg/sql.FinalizePlan(...)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:4671
github.com/cockroachdb/cockroach/pkg/sql/importer.distImport.func1({0xa322108, 0xc000d17680}, 0x71698ef?)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor_planning.go:115 +0x974
github.com/cockroachdb/cockroach/pkg/sql/importer.distImport({_, _}, {_, _}, _, _, {_, _, _}, {0x0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor_planning.go:119 +0x459
github.com/cockroachdb/cockroach/pkg/sql/importer.ingestWithRetry({_, _}, {_, _}, _, _, {_, _, _}, {0x0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_job.go:1258 +0x945
github.com/cockroachdb/cockroach/pkg/sql/importer.(*importResumer).Resume(0xc005754780, {0xa322108, 0xc003432d20}, {0x6efe5c0?, 0xc0021609a0})
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_job.go:290 +0x13ba
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0xc003e0eea0, 0xc0020f0900, {0xa2fb830, 0xc005754780}, {0xa322108, 0xc003432d20}, {0x6efe5c0, 0xc0021609a0}, 0xc0014e1638)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1534 +0x1a3
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc0020f0900, {0xa322108, 0xc00041b560}, {0x6efe5c0, 0xc0021609a0}, {0xa2fb830, 0xc005754780}, 0xc0046bcb60, {0xc0025b7f39, 0x7}, ...)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1535 +0xab2
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc0020f0900, {0xa322060, 0xc0037b3440}, {0xa2fb830, 0xc005754780}, 0xc0046bcb60, {0xc0025b7f39, 0x7}, {0xc000c583db, 0x5})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:467 +0x7ac
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func2({0xc0021a2c00?, 0xc005efac90?})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:385 +0x129
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470 +0x1f7
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:461 +0x61a
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestBackfillJobsInfoTable
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/194b1b11adb2bb86a3c51c4924595952/logTestBackfillJobsInfoTable2116041538
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillJobsInfoTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

upgrade/upgrades.TestBackfillJobsInfoTable [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9289030?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9289030?buildTab=artifacts#/) on master @ [143b63a6a27aeb286637dd2e5abddafdf0c51874](https://github.com/cockroachdb/cockroach/commits/143b63a6a27aeb286637dd2e5abddafdf0c51874):

Fatal error:

```
panic: unexpectedly no result routers in {0xc003f87f80 [] [] {[]} 0 local} [recovered]
	panic: unexpectedly no result routers in {0xc003f87f80 [] [] {[]} 0 local}

```
Stack: 

```
goroutine 54100 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x4afa45?, {0xa327a28, 0xc004a4ee40})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:229 +0x79
panic({0x6d13320, 0xc000463a30})
	GOROOT/src/runtime/panic.go:890 +0x262
github.com/cockroachdb/cockroach/pkg/sql/physicalplan.(*PhysicalPlan).GetResultTypes(0xc0052a9c00)
	github.com/cockroachdb/cockroach/pkg/sql/physicalplan/physical_plan.go:226 +0x1c5
github.com/cockroachdb/cockroach/pkg/sql/physicalplan.(*PhysicalPlan).EnsureSingleStreamOnGateway(0xc0052a9c00, {0xa327a28, 0xc0059f92f0})
	github.com/cockroachdb/cockroach/pkg/sql/physicalplan/physical_plan.go:494 +0x1af
github.com/cockroachdb/cockroach/pkg/sql.finalizePlanWithRowCount({0xa327a28, 0xc0059f92f0}, 0x0?, 0xc0052a9c00, 0x0?)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:4684 +0x94
github.com/cockroachdb/cockroach/pkg/sql.FinalizePlan(...)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:4671
github.com/cockroachdb/cockroach/pkg/sql/importer.distImport.func1({0xa327a28, 0xc0059f92f0}, 0x716dfcf?)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor_planning.go:115 +0x974
github.com/cockroachdb/cockroach/pkg/sql/importer.distImport({_, _}, {_, _}, _, _, {_, _, _}, {0x0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor_planning.go:119 +0x459
github.com/cockroachdb/cockroach/pkg/sql/importer.ingestWithRetry({_, _}, {_, _}, _, _, {_, _, _}, {0x0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_job.go:1258 +0x945
github.com/cockroachdb/cockroach/pkg/sql/importer.(*importResumer).Resume(0xc004020be0, {0xa327a28, 0xc000c669f0}, {0x6f02b80?, 0xc0029352b8})
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_job.go:290 +0x13ba
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0xc000fc1b00, 0xc0020c6900, {0xa301150, 0xc004020be0}, {0xa327a28, 0xc000c669f0}, {0x6f02b80, 0xc0029352b8}, 0xc006749638)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1562 +0x1a3
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc0020c6900, {0xa327a28, 0xc004a4f170}, {0x6f02b80, 0xc0029352b8}, {0xa301150, 0xc004020be0}, 0xc003cc5860, {0xc004d6b4c9, 0x7}, ...)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1563 +0xab2
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc0020c6900, {0xa327980, 0xc0029f0640}, {0xa301150, 0xc004020be0}, 0xc003cc5860, {0xc004d6b4c9, 0x7}, {0xc00500371b, 0x5})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:467 +0x7ac
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func2({0xc00416f400?, 0x203001?})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:385 +0x129
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470 +0x1f7
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:461 +0x61a
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestBackfillJobsInfoTable
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/194b1b11adb2bb86a3c51c4924595952/logTestBackfillJobsInfoTable1275620331
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillJobsInfoTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

upgrade/upgrades.TestBackfillJobsInfoTable [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9312657?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9312657?buildTab=artifacts#/) on master @ [61d1dce768ecf3b37e79f2621d3726d7ff4e95df](https://github.com/cockroachdb/cockroach/commits/61d1dce768ecf3b37e79f2621d3726d7ff4e95df):

Fatal error:

```
panic: unexpectedly no result routers in {0xc00baaa230 [] [] {[]} 0 local} [recovered]
	panic: unexpectedly no result routers in {0xc00baaa230 [] [] {[]} 0 local}

```
Stack: 

```
goroutine 9692071 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x1e19f2a?, {0x5cf39c8, 0xc00facb4d0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:229 +0x6a
panic({0x461f320, 0xc0125507d0})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/sql/physicalplan.(*PhysicalPlan).GetResultTypes(0xc0129582a0?)
	github.com/cockroachdb/cockroach/pkg/sql/physicalplan/physical_plan.go:226 +0xba
github.com/cockroachdb/cockroach/pkg/sql/physicalplan.(*PhysicalPlan).EnsureSingleStreamOnGateway(0xc0113e9580, {0x5cf39c8, 0xc01214c840})
	github.com/cockroachdb/cockroach/pkg/sql/physicalplan/physical_plan.go:494 +0x7c
github.com/cockroachdb/cockroach/pkg/sql.finalizePlanWithRowCount({0x5cf39c8, 0xc01214c840}, 0x0?, 0xc0113e9580, 0x0?)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:4687 +0x6e
github.com/cockroachdb/cockroach/pkg/sql.FinalizePlan(...)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:4674
github.com/cockroachdb/cockroach/pkg/sql/importer.distImport.func1({0x5cf39c8, 0xc01214c840}, 0x4a71ea4?)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor_planning.go:116 +0x5cf
github.com/cockroachdb/cockroach/pkg/sql/importer.distImport({_, _}, {_, _}, _, _, {_, _, _}, {0x0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor_planning.go:120 +0x2f9
github.com/cockroachdb/cockroach/pkg/sql/importer.ingestWithRetry({_, _}, {_, _}, _, _, {_, _, _}, {0x0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_job.go:1258 +0x76d
github.com/cockroachdb/cockroach/pkg/sql/importer.(*importResumer).Resume(0xc00df116d0, {0x5cf39c8, 0xc0113af1a0}, {0x480cbe0?, 0xc00b5f20d0})
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_job.go:290 +0xb26
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0xc00cf829c0?, 0x0?, {0x5ccd0b0?, 0xc00df116d0?}, {0x5cf39c8?, 0xc0113af1a0?}, {0x480cbe0?, 0xc00b5f20d0?}, 0xc012959860)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1562 +0xcc
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc002632480, {0x5cf39c8, 0xc00facb7d0}, {0x480cbe0, 0xc00b5f20d0}, {0x5ccd0b0, 0xc00df116d0}, 0xc00cf829c0, {0xc007029d68, 0x7}, ...)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1563 +0x8ed
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc002632480, {0x5cf3920, 0xc010e05580}, {0x5ccd0b0, 0xc00df116d0}, 0xc00cf829c0, {0xc007029d68, 0x7}, {0xc006c841d5, 0x5})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:467 +0x5a6
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func2({0xc00db79c00?, 0xc00ede8e40?})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:385 +0xfa
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470 +0x146
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:461 +0x43b
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestBackfillJobsInfoTable
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/194b1b11adb2bb86a3c51c4924595952/logTestBackfillJobsInfoTable2251634571
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99802 upgrade/upgrades: TestBackfillJobsInfoTable failed [C-test-failure O-robot T-sql-schema branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillJobsInfoTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

upgrade/upgrades.TestBackfillJobsInfoTable [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9312661?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9312661?buildTab=artifacts#/) on master @ [61d1dce768ecf3b37e79f2621d3726d7ff4e95df](https://github.com/cockroachdb/cockroach/commits/61d1dce768ecf3b37e79f2621d3726d7ff4e95df):

Fatal error:

```
panic: unexpectedly no result routers in {0xc004622850 [] [] {[]} 0 local} [recovered]
	panic: unexpectedly no result routers in {0xc004622850 [] [] {[]} 0 local}

```
Stack: 

```
goroutine 53138 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x4afa45?, {0xa32c068, 0xc005dea060})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:229 +0x79
panic({0x6d16cc0, 0xc0051caf70})
	GOROOT/src/runtime/panic.go:890 +0x262
github.com/cockroachdb/cockroach/pkg/sql/physicalplan.(*PhysicalPlan).GetResultTypes(0xc005563a80)
	github.com/cockroachdb/cockroach/pkg/sql/physicalplan/physical_plan.go:226 +0x1c5
github.com/cockroachdb/cockroach/pkg/sql/physicalplan.(*PhysicalPlan).EnsureSingleStreamOnGateway(0xc005563a80, {0xa32c068, 0xc001b1e8d0})
	github.com/cockroachdb/cockroach/pkg/sql/physicalplan/physical_plan.go:494 +0x1af
github.com/cockroachdb/cockroach/pkg/sql.finalizePlanWithRowCount({0xa32c068, 0xc001b1e8d0}, 0x0?, 0xc005563a80, 0x0?)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:4687 +0x94
github.com/cockroachdb/cockroach/pkg/sql.FinalizePlan(...)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:4674
github.com/cockroachdb/cockroach/pkg/sql/importer.distImport.func1({0xa32c068, 0xc001b1e8d0}, 0x7171cc4?)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor_planning.go:116 +0x974
github.com/cockroachdb/cockroach/pkg/sql/importer.distImport({_, _}, {_, _}, _, _, {_, _, _}, {0x0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor_planning.go:120 +0x459
github.com/cockroachdb/cockroach/pkg/sql/importer.ingestWithRetry({_, _}, {_, _}, _, _, {_, _, _}, {0x0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_job.go:1258 +0x945
github.com/cockroachdb/cockroach/pkg/sql/importer.(*importResumer).Resume(0xc0068dc820, {0xa32c068, 0xc00305dec0}, {0x6f064e0?, 0xc000cc2328})
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_job.go:290 +0x13ba
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0xc0061493e0, 0xc002694d80, {0xa305790, 0xc0068dc820}, {0xa32c068, 0xc00305dec0}, {0x6f064e0, 0xc000cc2328}, 0xc0053d5638)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1562 +0x1a3
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc002694d80, {0xa32c068, 0xc005dea360}, {0x6f064e0, 0xc000cc2328}, {0xa305790, 0xc0068dc820}, 0xc004445860, {0xc002289749, 0x7}, ...)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1563 +0xab2
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc002694d80, {0xa32bfc0, 0xc000a30880}, {0xa305790, 0xc0068dc820}, 0xc004445860, {0xc002289749, 0x7}, {0xc002289a3b, 0x5})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:467 +0x7ac
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func2({0xc005c15c00?, 0x203001?})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:385 +0x129
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470 +0x1f7
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:461 +0x61a
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestBackfillJobsInfoTable
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/194b1b11adb2bb86a3c51c4924595952/logTestBackfillJobsInfoTable3092508749
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99802 upgrade/upgrades: TestBackfillJobsInfoTable failed [C-test-failure O-robot T-sql-schema branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillJobsInfoTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

upgrade/upgrades.TestBackfillJobsInfoTable [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9336053?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9336053?buildTab=artifacts#/) on master @ [37ac1e5f751d981ecced8f2bf94f12cb2ab7e150](https://github.com/cockroachdb/cockroach/commits/37ac1e5f751d981ecced8f2bf94f12cb2ab7e150):

Fatal error:

```
panic: unexpectedly no result routers in {0xc000fb6310 [] [] {[]} 0 local} [recovered]
	panic: unexpectedly no result routers in {0xc000fb6310 [] [] {[]} 0 local}

```
Stack: 

```
goroutine 54071 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x695fb20?, {0xa351d88, 0xc001e49e60})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:229 +0x79
panic({0x6d35940, 0xc0057db000})
	GOROOT/src/runtime/panic.go:890 +0x262
github.com/cockroachdb/cockroach/pkg/sql/physicalplan.(*PhysicalPlan).GetResultTypes(0xc003989680)
	github.com/cockroachdb/cockroach/pkg/sql/physicalplan/physical_plan.go:226 +0x1c5
github.com/cockroachdb/cockroach/pkg/sql/physicalplan.(*PhysicalPlan).EnsureSingleStreamOnGateway(0xc003989680, {0xa351d88, 0xc002137da0})
	github.com/cockroachdb/cockroach/pkg/sql/physicalplan/physical_plan.go:494 +0x1af
github.com/cockroachdb/cockroach/pkg/sql.finalizePlanWithRowCount({0xa351d88, 0xc002137da0}, 0x0?, 0xc003989680, 0x0?)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:4687 +0x94
github.com/cockroachdb/cockroach/pkg/sql.FinalizePlan(...)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:4674
github.com/cockroachdb/cockroach/pkg/sql/importer.distImport.func1({0xa351d88, 0xc002137da0}, 0x7191310?)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor_planning.go:116 +0x974
github.com/cockroachdb/cockroach/pkg/sql/importer.distImport({_, _}, {_, _}, _, _, {_, _, _}, {0x0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor_planning.go:120 +0x459
github.com/cockroachdb/cockroach/pkg/sql/importer.ingestWithRetry({_, _}, {_, _}, _, _, {_, _, _}, {0x0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_job.go:1258 +0x945
github.com/cockroachdb/cockroach/pkg/sql/importer.(*importResumer).Resume(0xc002371f90, {0xa351d88, 0xc003d245a0}, {0x6f254e0?, 0xc0009d6118})
	github.com/cockroachdb/cockroach/pkg/sql/importer/import_job.go:290 +0x13ba
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0xc001140ea0, 0xc00260c480, {0xa32b3d8, 0xc002371f90}, {0xa351d88, 0xc003d245a0}, {0x6f254e0, 0xc0009d6118}, 0xc0061d55f0)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1562 +0x1a3
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc00260c480, {0xa351d88, 0xc005a1c1e0}, {0x6f254e0, 0xc0009d6118}, {0xa32b3d8, 0xc002371f90}, 0xc0051a7860, {0xc004972fe9, 0x7}, ...)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1563 +0xab2
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc00260c480, {0xa351ce0, 0xc00436bc40}, {0xa32b3d8, 0xc002371f90}, 0xc0051a7860, {0xc004972fe9, 0x7}, {0xc002a5fd8b, 0x5})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:470 +0x7ac
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func2({0xa351d88, 0xc001e49e60})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:388 +0x198
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470 +0x1f7
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:461 +0x61a
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestBackfillJobsInfoTable
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/194b1b11adb2bb86a3c51c4924595952/logTestBackfillJobsInfoTable117980394
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99802 upgrade/upgrades: TestBackfillJobsInfoTable failed [C-test-failure O-robot T-sql-schema branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillJobsInfoTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/upgrade/upgrades/backfill_jobs_info_table_migration_test.go
+++ b/pkg/upgrade/upgrades/backfill_jobs_info_table_migration_test.go
@@ -53,6 +53,17 @@ func TestBackfillJobsInfoTable(t *testing.T) {
 	r := tc.Server(0).JobRegistry().(*jobs.Registry)
 	sqlDB := sqlutils.MakeSQLRunner(tc.ServerConn(0))
 	defer tc.Stopper().Stop(ctx)
+	r.TestingResumerCreationKnobs = map[jobspb.Type]func(raw jobs.Resumer) jobs.Resumer{}
+	jobspb.ForEachType(func(typ jobspb.Type) {
+		// The upgrade creates migration and schemachange jobs, so we do not
+		// need to create more. We should not override resumers for these job types,
+		// otherwise the upgrade will hang.
+		if typ != jobspb.TypeMigration && typ != jobspb.TypeSchemaChange {
+			r.TestingResumerCreationKnobs[typ] = func(r jobs.Resumer) jobs.Resumer {
+				return &fakeResumer{}
+			}
+		}
+	}, false)
 
 	createJob := func(id jobspb.JobID, details jobspb.Details, progress jobspb.ProgressDetails) *jobs.Job {
 		defaultRecord := jobs.Record{

